bin/docker_utils.sh

```diff
@@ -80,3 +80,37 @@ function ensure_container_running {
     done
     echo "$RUNNING"
 }
+
+# get all matching images, will return image IDs in the order of most recent first
+function get_matching_image_ids {
+    IMAGE_NAME="$1"
+    IMAGES=$(docker images | grep -F $IMAGE_NAME | awk '{ print $3 }')
+    echo $IMAGES
+}
+
+# remove all but the most recent release image
```

Comment:
I can't ascertain at a glance: does this mean the image needed for a faster potential rollback would be retained, in addition to the image for the current / upcoming deployment?

Reply:
Good thought - no, it wouldn't, but the rollback deployment will pull that image before stopping the existing container.  However, it would then try to clean it up by this logic (which would fail) and then mark the deployment as "failed" even though it succeeded.  We could keep the last 2 images to account for a potential rollback without altering any other deployment logic.

Reply:
Cool, keeping the last 2 images indeed seems robust, efficient, and simple for the case of rollback.

Reply:
[7b1a75e](https://github.com/broadinstitute/single_cell_portal_core/pull/1623/commits/7b1a75e887a306db105ae91524897e3f05ce8aaf)